backend/src/main/java/ca/bc/gov/app/service/client/ClientService.java

```diff
@@ -141,6 +142,76 @@ public Mono<ClientDetailsDto> getClientDetails(
             "Unable to process request. This sole proprietor is not owned by a person"
         )));
   }
+  
+  public Mono<ForestClientDetailsDto> getClientDetailsByClientNumber(
+      String clientNumber,
+      List<String> groups
+  ) {
+      log.info("Loading details for {} for user role {}", clientNumber, groups.toString());
+
+      return legacyService
+          .searchByClientNumber(clientNumber, groups)
+          .flatMap(forestClientDetailsDto -> {
+              String corpRegnNmbr = forestClientDetailsDto.corpRegnNmbr();
+
+              if (corpRegnNmbr == null || corpRegnNmbr.isEmpty()) {
+                  log.info("Corporation registration number not provided. Returning legacy details.");
+                  return Mono.just(forestClientDetailsDto);
+              }
+
+              log.info("Retrieved corporation registration number: {}", corpRegnNmbr);
+
+              Mono<BcRegistryDocumentDto> documentMono = bcRegistryService
+                  .requestDocumentData(corpRegnNmbr)
+                  .next();
```

Comment:
Instead of stopping here, you can call `.flatMap` next and call your method inside of it to keep the chain of execution